model/src/main/java/de/danoeh/antennapod/model/feed/Transcript.java

```diff
@@ -0,0 +1,43 @@
+package de.danoeh.antennapod.model.feed;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+public class Transcript {
+
+    private final TreeMap<Long, TranscriptSegment> segmentsMap = new TreeMap<>();
+
+    public void addSegment(TranscriptSegment segment) {
+        segmentsMap.put(segment.getStartTime(), segment);
+    }
+
+    public void updateSegment(Long timeOld, Long timeNew) {
```

Comment:
Hmm, now that I'm thinking about it more (and looking at your repo with the future PRs), I'm not 100% convinced that we should use this method. For now, I would rather remove it from this PR (it's unused anyway). The reason is that the future code currently modifies an existing transcript (using this method and others). The problem with this is that we might want to display the transcript in two different locations ("subtitle" and transcript screen). So if we modify the original transcript object, that could lead to unexpected results. Instead, I would rather have screens create a copy of the transcript that is broken up to fit the screen.

Anyway, this discussion is a bit out of scope for this PR (which only does the parsing), so I would for now remove the unused methods here and discuss this when actually working on displaying.

Reply:
Looking forward to discussing this in the next PR.  Removing `updateSegment()`

Reply:
Is the "remove" method also currently unused? Then I would remove that for now as well